### ISSUE TITLE
Group Commits

### ISSUE THREAD

**Issue by SHADOWELITE7** [sublimehq/sublime_merge#857]
**Problem description**

Group Commits can be useful for larger projects. You individually click on a file then click on another file. You can also click on which files can go into Gitignore or full deletion.

**Preferred solution**

A similar way to what GH Desktop looks like. In the untracked files tab for merge. Next to each file should show you a checkbox so you can Click one at a time or use a keybinding.

**Alternatives**

GH Desktop may be the alternative because it already has this feature. But I think Merge should have this because its faster than GH Desktop.

**Screenshots**

an example from GH Desktop
![image](https://user-images.githubusercontent.com/33847580/89539537-df1f7300-d7c9-11ea-8b26-018d4e32031d.png)

**Comment by stinos** [comment]
Sorry this is rather confiusing, especially for people who've never used GH Desktop. Your title mentions 'group commits' but in the description you're talking about grouping files, it seems. Can you clarify what exactly this is supposed to do, i.e. what are you going to do with the files which have been checked? Stage?

**Comment by kpowick** [comment]
@SHADOWELITE7  SM does allow you to select groups of new/changed files to commit.  The Git concept is staging, which adds files to the Git Index.  It is these staged files that you commit.

I think your confusion comes from GH Desktop itself, which does not support the concept of staging.  I have little experience with GH Desktop, but it seems more focused on working with Github rather than supporting a comprehensive set of Git features.

**Comment by SHADOWELITE7** [comment]
Sorry I haven't responded. @kpowick I think that's what I meant. but sorry for the confusion with GH Desktop. 

With the new update of Merge. I would click stage to manually stage one file. Couldn't figure out how to stage multiple files. Though usually, it shows the stage all button. It didn't have stage all when I was committing 1000+ files.

**Comment by kpowick** [comment]
You are correct about SM behavior when there are many files.  I downloaded the Linux project (65000+ files) to see how SM would handle it.

On the "_Summary_" tab, the only option is "_Show Files_", which activates the "_Files_" tab and lists all the files in the project.

<img width="1301" alt="Screen Shot 2020-08-10 at 09 44 20" src="https://user-images.githubusercontent.com/1646753/89790161-89133e00-daef-11ea-8212-cd1c8bfcadbc.png">

There is a context menu (right-click) option to "_Stage All_", but that seems to be the only option for file selection.  There is no way to select groups of files from the list via shift-click for a range of files, or cmd-click for individual files.  I think such functionality, or check boxes as you suggested, would be a worthwhile enhancement.

<img width="573" alt="Screen Shot 2020-08-10 at 09 45 04" src="https://user-images.githubusercontent.com/1646753/89790051-63863480-daef-11ea-8243-e5ebb568fb80.png">

**Comment by SHADOWELITE7** [comment]
I see, thanks for the image. I haven't tried to right click on the unstaged files before. Next time I should try that.
